src/csharp/Microsoft.Spark.Worker.UnitTest/TaskRunnerTests.cs

```diff
@@ -40,6 +40,11 @@ public void TestTaskRunner()
                 System.IO.Stream inputStream = serverSocket.InputStream;
                 System.IO.Stream outputStream = serverSocket.OutputStream;
 
+                if (payloadWriter.Version.Major == 3 && payloadWriter.Version.Minor == 2)
+                {
+                    int pid = PayloadReader.ReadInt(inputStream);
+                }
+
```

Comment:
I feel like this should belong to the `SimpleWorkerTests` instead.   What happens if you change line 25 of this file to Create a V3_2_0 `PayloadWriter`?

Reply:
It causes the test to hang. If we move the check to `SimpleWorkerTests` we would have to move socket read and write streaming logic out of `TestTaskRunnerReadWrite`.

Reply:
That causes the test to hang. Moving it upto `SimpleWorkerTests`, required a little refactoring of tests.